### PR TITLE
Add parser for protolint

### PR DESCRIPTION
**Pull request by davidhart82**

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.regex.Matcher;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.RegexpLineParser;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Parser for ProtoLint.
+ */
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java]: Optionally add your name as \@author.

File: src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class ProtoLintParser extends RegexpLineParser {
+
+    private static final long serialVersionUID = -8347619672754062010L;
+
+    private static final String PROTOLINT_PATTERN = "^\\[([^:]+):(\\d+):(\\d+)\\] (.+)$";
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java]: Can you please use named groups? It makes the regexp more readable, see [EclipseMavenParser](https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/EclipseMavenParser.java#L29) for an example.

File: src/test/java/edu/hm/hafner/analysis/parser/ProtoLintParserTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import static edu.hm.hafner.analysis.assertj.IssuesAssert.assertThat;
+
+import edu.hm.hafner.analysis.AbstractParserTest;
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import edu.hm.hafner.analysis.assertj.SoftAssertions;
+
+class ProtoLintParserTest extends AbstractParserTest {
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/ProtoLintParserTest.java]: Please add a JavaDoc for the class.

File: src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java (hunk 1 above)
Comment (davidhart82) [reply src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java]: Done

File: src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java (hunk 2 above)
Comment (davidhart82) [reply src/main/java/edu/hm/hafner/analysis/parser/ProtoLintParser.java]: Done

File: src/test/java/edu/hm/hafner/analysis/parser/ProtoLintParserTest.java (hunk 3 above)
Comment (davidhart82) [reply src/test/java/edu/hm/hafner/analysis/parser/ProtoLintParserTest.java]: Done

**Comment by davidhart82** [comment]
Hey, what do I have to do to integrate this into the warnings-ng plugin?  Thanks.

**Comment by uhafner** [comment]
Write a small descriptor class in the style of [EsLint](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/java/io/jenkins/plugins/analysis/warnings/EsLint.java) (and add a new test case in [ParsersITest](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java#L114).

**Pull request merged**